### ISSUE TITLE
Suggestion: Co-ordinate sync with reality [OPT-IN]

### ISSUE THREAD

**Issue by goteusz-maszyk** [louis-e/arnis#29]
Instead of starting at 0,0 the generator would start at the position that would allow generating a few tows next to each other, in correct geographical positions. As many places would end up far away, It would log the position when finished

**Comment by daniil2327** [comment]
Just generate the whole Earth!

**Comment by louis-e** [comment]
I like the idea of mapping the real coordinates to Minecraft coordinates, I already added it to the README ToDo section. What type of implementation would you prefer? A hosted website for an easy coordinate conversion (including an easy bbox finder) or extending the command line tool with a conversion feature?

**Comment by scd31** [comment]
> What type of implementation would you prefer?

I've been thinking about this a bit and I think the best way would be to encode metadata in the level.dat file, via some additional NBT tags. Then a CLI could read that metadata when doing scaling. It would eliminate the need to have the user input world details manually into the CLI (or into a web utility).

Semi-related, but eventually I'd like to write a tile server, similar to Dynmap, but optimized for Arnis. One of the things I'd like for it to be able to do is read this same metadata and present it in the web UI (possibly with a converter from real lat/lon to MC coords)

**Comment by cherbert** [comment]
As great as this tool is it's pretty useless without this feature being a priority right now. Every time you go back and generate a new area it renders it at 0,0,0. It makes it impossible to generate large worlds.

**Comment by amir16yp** [comment]
i would like to add, that you can use journeymap mod to render all the chunks (i think it calls it "automap" or something like that) and then if you are in creative mode you can just right click anywhere on the map and teleport

**Comment by cherbert** [comment]
> i would like to add, that you can use journeymap mod to render all the chunks (i think it calls it "automap" or something like that) and then if you are in creative mode you can just right click anywhere on the map and teleport. this can serve as a temporary solution for when you need to get to somewhere specific and you dont know where it is

Not sure how this is relevant to this conversation?

**Comment by amir16yp** [comment]
> > i would like to add, that you can use journeymap mod to render all the chunks (i think it calls it "automap" or something like that) and then if you are in creative mode you can just right click anywhere on the map and teleport. this can serve as a temporary solution for when you need to get to somewhere specific and you dont know where it is
> 
> Not sure how this is relevant to this conversation?

OP is asking about coordinate sync i.e a way to convert IRL coordinates into minecraft ones. im just saying if you need to get somewhere specific in the map (which is the use case) you can use journeymap. that is all

**Comment by cherbert** [comment]
> > > i would like to add, that you can use journeymap mod to render all the chunks (i think it calls it "automap" or something like that) and then if you are in creative mode you can just right click anywhere on the map and teleport. this can serve as a temporary solution for when you need to get to somewhere specific and you dont know where it is
> > 
> > 
> > Not sure how this is relevant to this conversation?
> 
> OP is asking about coordinate sync i.e a way to convert IRL coordinates into minecraft ones. im just saying if you need to get somewhere specific in the map (which is the use case) you can use journeymap. that is all

OP didn't ask that at all. Your reply is irrelevant to the question.

**Comment by LonghiTW** [comment]
The earth is round, you can't constantly generate surrounding areas without projection.
If you want to map the real coordinates to Minecraft coordinates, it would be a good way to use the projection method of the Build The Earth project or even working with BTE.

**Comment by george112n** [comment]
Indeed, Build the Earth use a modified dymaxion projection to generate the Earth. BTE experimented with using a flat projection whereby geographic coordinates could be converted to minecraft coordinates by multiplying by a simple scalar. I believe the scalar was 10^x, where I can't remember what x was. The problem was that even at relatively average latitudes, like 50 degrees north, a square was distorted into being a parallelogram with fairly large angles.

I've attached an image of the projection which BTE now uses. It's a dymaxion projection which was specifically designed to reduce both distortion of shape and size.

The only thing is, the in game cardinal directions no longer align with the geographic directions. i.e, something north in real life doesn't end up being north in game. The projection was rotated to try to retain some alignment of cardinal directions.

![image](https://github.com/user-attachments/assets/0b4d5b1e-da0f-48dc-b325-58499d60ac4d)
Each color band adds 0.1 to the distortion. First red band: 1.1 Blocks : 1 meter

To get somewhere specific on the map BTE has a command whereby someone does /tpll [latitude],[longitude] and it will take them to that place on the map and generate it there and then, without the full building shells.

In conclusion, BTE have already gone through the process of finding a good projection and this was the best it could come up with.

**Comment by george112n** [comment]
Also further to this, generating land on a projection allows you to do a lot of cool stuff with it. To name a few examples, I'm current developing a project where **stars** and other celestial bodies appear in the sky at the alt/az that they would if the player was at that position in real life, and it's all based on converting the player's current position to a geographical position. This also includes the sun and moon position. I'm planning to do a similar thing with the weather. Although this all requires a client side mod.

**Comment by cherbert** [comment]
Is this on the radar anytime soon? It's the only thing holding me back from being able to use this tool.

**Comment by SmartDarwin** [comment]
@louis-e  may I please suggest the ability to not include the buildings as an option and the ability to rotate the world by a certain degree. The small location I would like to try is at North -+45deg which makes all the roads and buildings diagonal and not straight lines. Thank you in advance.
